internal/services/apimanagement/api_management_policy_fragment_resource.go

```diff
@@ -0,0 +1,232 @@
+				return nil, err
+			}
+
+			resp, err := client.Get(ctx, *id, policyfragment.GetOperationOptions{
+				Format: pointer.To(policyfragment.PolicyFragmentContentFormatXml),
+			})
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return nil, fmt.Errorf("Api Management Policy Fragment %q was not found in Api Management instance %q in Resource Group %q", id.PolicyFragmentName, id.ServiceName, id.ResourceGroupName)
+				}
+
+				return nil, fmt.Errorf("retrieving Api Management Policy Fragment %q (Api Management: %q, Resource Group %q): %+v", id.PolicyFragmentName, id.ServiceName, id.ResourceGroupName, err)
+			}
+
+			d.Set("format", string(policyfragment.PolicyFragmentContentFormatXml))
+
+			return []*pluginsdk.ResourceData{d}, nil
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": schemaz.SchemaApiManagementChildName(),
+
+			"api_management_id": commonschema.ResourceIDReferenceRequiredForceNew(&apimanagementservice.ServiceId{}),
+
+			"format": {
+				Type:     pluginsdk.TypeString,
+				ForceNew: true,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(policyfragment.PolicyFragmentContentFormatRawxml),
+					string(policyfragment.PolicyFragmentContentFormatXml),
+				}, false),
+				Default: policyfragment.PolicyFragmentContentFormatXml,
+			},
+
+			"value": {
+				Type:             pluginsdk.TypeString,
+				Required:         true,
+				DiffSuppressFunc: XmlWithDotNetInterpolationsDiffSuppress,
+			},
+
+			"description": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceApiManagementPolicyFragmentCreate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).ApiManagement.PolicyFragmentClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	apiManagementId, err := apimanagementservice.ParseServiceID(d.Get("api_management_id").(string))
+	if err != nil {
+		return err
+	}
+	id := policyfragment.NewPolicyFragmentID(apiManagementId.SubscriptionId, apiManagementId.ResourceGroupName, apiManagementId.ServiceName, d.Get("name").(string))
+	format := policyfragment.PolicyFragmentContentFormat(d.Get("format").(string))
+	if d.IsNewResource() {
+		opts := policyfragment.DefaultGetOperationOptions()
+		opts.Format = &format
+		existing, err := client.Get(ctx, id, opts)
+		if err != nil {
+			if !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %s", id, err)
+			}
+		}
+
+		if !response.WasNotFound(existing.HttpResponse) {
+			return tf.ImportAsExistsError("azurerm_api_management_policy_fragment", id.ID())
+		}
+	}
```

Comment:
`d.IsNewResource()` is only applicable if the Create and Update methods are combined, which is a pattern we're moving away from. So this can be removed 
```suggestion

	opts := policyfragment.DefaultGetOperationOptions()
	opts.Format = &format
	existing, err := client.Get(ctx, id, opts)
	if err != nil {
		if !response.WasNotFound(existing.HttpResponse) {
			return fmt.Errorf("checking for presence of existing %s: %s", id, err)
		}
	}

	if !response.WasNotFound(existing.HttpResponse) {
		return tf.ImportAsExistsError("azurerm_api_management_policy_fragment", id.ID())
	}

```